### PR TITLE
Fix EZP-24040: Date/Time attributes are not timezone friendly

### PR DESCRIPTION
**Pull request by yannickroger**
Link: https://jira.ez.no/browse/EZP-24040

## Description
This PR contains two fixes for Date and Time attributes
### Time
The time is now stored and displayed in UTC. It doesn't have to be timezone aware are it's just a plain data.

### DateAndTime
The API stores DateAndTime fields in UTC format and doesn't store the timezone. This patch converts back and forth UTC to local time in order to display something meaningful to the user.

## Test
(Lots of) manual test. Updated Unit test. It is at the moment not possible to automate more testing because of locale limitation in the headless browser used to run our unit tests.

## TODO
- [x] Cleanup

### CODE REVIEW

File: Resources/public/js/views/fields/ez-dateandtime-editview.js [hunk 1]
```diff
@@ -264,6 +264,19 @@ YUI.add('ez-dateandtime-editview', function (Y) {
         },
 
         /**
+         * Returns a timestamps in UTC
+         *
+         * @protected
+         * @method _getUtcTimeStamp
+         * @param {Integer} localizedTimestamp
```
Comment (dpobel) [inline Resources/public/js/views/fields/ez-dateandtime-editview.js]: in JS, it's `{Number}` not `{Integer}`

File: Resources/public/js/views/fields/ez-dateandtime-editview.js [hunk 2]
```diff
@@ -264,6 +264,19 @@ YUI.add('ez-dateandtime-editview', function (Y) {
...
+         * @return {Integer} Timestamp converted to UTC
```
Comment (dpobel) [inline Resources/public/js/views/fields/ez-dateandtime-editview.js]: you should mention that it's in millisecond.

File: Resources/public/js/views/fields/ez-time-editview.js [hunk 3]
```diff
@@ -101,14 +101,17 @@ YUI.add('ez-time-editview', function (Y) {
...
             if ( field && field.fieldValue ) {
                 if (!this.get('supportsTimeInput') && !def.fieldSettings.useSeconds) {
-                    time =  Y.Date.format(new Date(field.fieldValue * 1000), {format:"%R"});
+                    options = {hour: 'numeric', minute: 'numeric', timeZone:'UTC'};
                 } else {
-                    time =  Y.Date.format(new Date(field.fieldValue * 1000), {format:"%T"});
+                    options = {hour: 'numeric', minute: 'numeric', second: 'numeric', timeZone:'UTC'};
                 }
+
+                time = new Date(field.fieldValue * 1000).toLocaleTimeString(undefined, options);
```
Comment (dpobel) [inline Resources/public/js/views/fields/ez-time-editview.js]: unfortunately, the support for the parameters of `toLocaleTimeString` is rather low (Safari does not support that it seems plus several mobile agent), so we cannot use it

**Comment by dpobel** [comment]
+1

**Comment by dpobel** [comment]
(The tests are are  failing on Travis, it seems like it's because of a missing dependency)

ping @mhyndle @andrerom

**Comment by mhyndle** [comment]
+1

**Comment by yannickroger** [comment]
@dpobel @mhyndle I fixed the unit test and while at it I added an extra commit fixing the error running `grunt test` because of locale missmatch.
Is it still ok for you guys ?

File: Tests/js/views/fields/ez-dateandtime-view.html [hunk 4]
```diff
@@ -29,11 +29,11 @@
         filter: loaderFilter,
         modules: {
             "ez-dateandtime-view": {
-                requires: ['ez-fieldview'],
+                requires: ['ez-fieldview', 'datatype-date-format'],
                 fullpath: "../../../../Resources/public/js/views/fields/ez-dateandtime-view.js"
             },
             "ez-fieldview": {
-                requires: ['ez-templatebasedview'],
+                requires: ['ez-templatebasedview', 'datatype-date-format'],
```
Comment (dpobel) [inline Tests/js/views/fields/ez-dateandtime-view.html]: not needed :)

**Comment by dpobel** [comment]
besides the last inline comment, +1 (if Travis agrees :p)

**Comment by mhyndle** [comment]
+1 confirmed

**Pull request merged**